llvm/lib/ExecutionEngine/JITLink/DWARFRecordSectionSplitter.cpp

```diff
@@ -33,35 +34,28 @@ Error DWARFRecordSectionSplitter::operator()(LinkGraph &G) {
            << "...\n";
   });
 
-  DenseMap<Block *, LinkGraph::SplitBlockCache> Caches;
-
-  {
-    // Pre-build the split caches.
-    for (auto *B : Section->blocks())
-      Caches[B] = LinkGraph::SplitBlockCache::value_type();
-    for (auto *Sym : Section->symbols())
-      Caches[&Sym->getBlock()]->push_back(Sym);
-    for (auto *B : Section->blocks())
-      llvm::sort(*Caches[B], [](const Symbol *LHS, const Symbol *RHS) {
-        return LHS->getOffset() > RHS->getOffset();
-      });
-  }
+  std::vector<Block *> Blocks;
+  for (auto *B : Section->blocks())
+    Blocks.push_back(B);
```

Comment:
Why is this copy operation necessary?

Reply:
Splitting will introduce new blocks, which would invalidate iterators in the range-based `for` below.

It looks like this has removed the pre-built caching though? I don't think we want to do that (or at least it couldn't be upstreamed) -- it'd affect performance for performance-sensitive operations like splitting eh-frames.